documentation: PostgresVectorStore documentation

Itzaprado opened the pull request.

Itzaprado committed:
documentation: add README and cloudsql_pg_store example

Itzaprado committed:
documentation: add how-to for vectorStore interface

Itzaprado committed:
documentation: updating example

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsq_pg.ipynb:
Update to "google_cloudsql_pg.ipynb"

averikitsch commented on examples/src/stores/cloudsql_pg_store.ts:
Move this file under examples/src/indexes/vector_stores

averikitsch commented on examples/src/stores/cloudsql_pg_store.ts:
```diff
@@ -0,0 +1,75 @@
+  contentColumn: "my_content",
+  embeddingColumn: "my_embedding",
+  metadataColumns: ["page", "source"],
+  metadataJsonColumn: "my_metadata",
```
Lets just keep metadataColumns

averikitsch commented on examples/src/stores/cloudsql_pg_store.ts:
```diff
@@ -0,0 +1,75 @@
+};
+
+// Vector store table initilization
+await engine.initVectorstoreTable("my_vector_store_table", 768);
```
Let's set up the metadatacolumns in the table as well

averikitsch commented on libs/langchain-google-cloud-sql-pg/README.md:
```diff
@@ -0,0 +1,85 @@
+* The package creates a shared connection pool to connect to Google Cloud Postgress databases utilizing different ways for authentication such as IAM, user and password authorization.
+* Store metadata in columns instead of JSON, resulting in significant performance improvements.
+
+##  Quick Start
```
```suggestion
##  Before you begin
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/README.md:
```diff
@@ -0,0 +1,85 @@
+# LangChain package for CloudSQL for Postgress
```
```suggestion
# @langchain/google-cloud-sql-pg
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/README.md:
````diff
@@ -0,0 +1,85 @@
+### Installation
+
+```bash
+$ yarn add @langchain/langchain-google-cloud-sql-pg
````
THis will most likely be released as @langchain/google-cloud-sql-pg

averikitsch replied:
Please update all locations

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsq_pg.ipynb:
```diff
@@ -0,0 +1,326 @@
+   "source": [
+    "## Manage Vector Store\n",
+    "\n",
+    "### Add items to vector store\n",
```
Update items to Documents throughout the doc

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsq_pg.ipynb:
```diff
@@ -0,0 +1,326 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Query vector store\n",
```
```suggestion
    "## Search for documents\n",
```

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsq_pg.ipynb:
```diff
@@ -0,0 +1,326 @@
+    "PASSWORD=\"your-database-password\"\n",
+    "EMAIL=\"your-email-account\"\n",
+    "PG_URL=\"postgresql+asyncpg://your-db-user:your-db-password@your-host:5432/your-db-name\"\n",
+    "HOST=\"your-host\""
```
Let's remove variables we aren't using.

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsq_pg.ipynb:
```diff
@@ -0,0 +1,326 @@
+    "7.  [Add a user to the database](https://cloud.google.com/sql/docs/postgres/create-manage-users)\n"
+   ]
+  },
+  {
```
We probably need to add a set to setup auth https://github.com/googleapis/langchain-google-alloydb-pg-python/blob/edd2283a1fcbd6d0c4dbb6884cc32e44bf4faa22/docs/vector_store.ipynb#L119

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsq_pg.ipynb:
```diff
@@ -0,0 +1,326 @@
+     "languageId": "typescript"
+    }
+   },
+   "outputs": [],
```
We need a setup to install the dependencies

Itzaprado committed:
refactor: vectorStore vectorStore
[ ] rename files, move files to right folder
[ ] change some titles and references
[ ] Add authentication and set project sections

averikitsch commented on libs/langchain-google-cloud-sql-pg/README.md:
```diff
@@ -0,0 +1,85 @@
+ pgArgs
+);
+
+await engine.init_vectorstore_table("my-table", 768);
```
```suggestion
await engine.initVectorstoreTable("my-table", 768);
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/README.md:
```diff
@@ -0,0 +1,85 @@
+    password: "password"
+}
+
+const engine: PostgresEngine = await PostgresEngine.from_instance(
```
```suggestion
const engine: PostgresEngine = await PostgresEngine.fromInstance(
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/README.md:
```diff
@@ -0,0 +1,85 @@
+-   similaritySearch
+-   and others.
+
+<!-- TODO: add reference to the how to doc -->
```
```suggestion
See the full [Vector Store](https://js.langchain.com/docs/integrations/vectorstores/google_cloudsql_pg) tutorial.
```

averikitsch commented on examples/src/indexes/vector_stores/cloudsql_pg_store.ts:
```diff
@@ -0,0 +1,89 @@
+  peArgs
+);
+
+const pvectorArgs: PostgresVectorStoreArgs = {
+  idColumn: "uuid",
+  contentColumn: "my_content",
+  embeddingColumn: "my_embedding",
```
```suggestion
const pvectorArgs: PostgresVectorStoreArgs = {
```

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsql_pg.ipynb:
```diff
@@ -0,0 +1,342 @@
+   },
+   "outputs": [],
+   "source": [
+    "// deleting an item\n",
```
```suggestion
    "// deleting a document\n",
```

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsql_pg.ipynb:
```diff
@@ -0,0 +1,342 @@
+    "const id1 = ids[0];\n",
+    "await vectorStore.delete({ ids: [id1] });\n",
+    "\n",
+    "// deleting more than one item\n",
```
```suggestion
    "// deleting more than one document\n",
```

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsql_pg.ipynb:
````diff
@@ -0,0 +1,342 @@
+    "### Set Your Google Cloud Project\n",
+    "\n",
+    "Set your Google Cloud project ID to leverage Google Cloud resources locally:\n",
+    "*   ```gcloud config set project YOUR-PROJECT-ID```\n",
````
Update this to be a code cell

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsql_pg.ipynb:
```diff
@@ -0,0 +1,342 @@
+    ");\n",
+    "\n",
+    "const pvectorArgs: PostgresVectorStoreArgs = {\n",
+    "  idColumn: \"uuid\",\n",
+    "  contentColumn: \"my_content\",\n",
+    "  embeddingColumn: \"my_embedding\",\n",
+    "  metadataColumns: [\"page\", \"source\"],\n",
+    "  metadataJsonColumn: \"my_metadata\",\n",
```
```suggestion
    "  metadataColumns: [\"page\", \"source\"],\n",
```

averikitsch commented on docs/core_docs/docs/integrations/vectorstores/google_cloudsql_pg.ipynb:
```diff
@@ -0,0 +1,342 @@
+    "const embeddingService = new SyntheticEmbeddings({ vectorSize: 768 });\n",
+    "\n",
+    "// PostgresVectorStore instantiation\n",
+    "const vectorStore = await PostgresVectorStore.create(\n",
```
Can you add the metadataColumn args to the VectorStore?

Itzaprado committed:
refactor: documentation
[ ] rename with right methods in README
[ ] add missing args to some references
[ ] remove unnecessary code
